clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/AbstractKafkaConsumerConfigProperties.java

```diff
@@ -0,0 +1,162 @@
+        return containsMinimalConfiguration(commonClientConfig) || containsMinimalConfiguration(consumerConfig);
+    }
+
+    /**
+     * Gets the Kafka consumer configuration. This includes changes made in {@link #adaptConfiguration(Map)}. The
+     * returned map will contain a property {@code client.id} which will be set to a unique value containing the already
+     * set client id or alternatively the value set via {@link #setDefaultClientIdPrefix(String)}, the given
+     * consumerName and a newly created UUID.
+     *
+     * Note: This method should be called for each new consumer, ensuring that a unique client id is used.
+     *
+     * @param consumerName A name for the consumer to include in the added {@code client.id} property.
+     * @return a copy of the consumer configuration with the applied properties or an empty map if neither a consumer
+     *         client configuration was set with {@link #setConsumerConfig(Map)} nor common configuration properties
+     *         were set with {@link #setCommonClientConfig(Map)}.
+     * @throws NullPointerException if consumerName is {@code null}.
+     */
+    public final Map<String, String> getConsumerConfig(final String consumerName) {
+        Objects.requireNonNull(consumerName);
+
+        if (commonClientConfig == null && consumerConfig == null) {
+            return new HashMap<>();
+        }
+
+        final Map<String, String> newConfig = new HashMap<>();
+        if (commonClientConfig != null) {
+            newConfig.putAll(commonClientConfig);
+        }
+        if (consumerConfig != null) {
+            newConfig.putAll(consumerConfig);
+        }
+
+        adaptConfiguration(newConfig);
+
+        Optional.ofNullable(keyDeserializerClass).ifPresent(serializerClass -> overrideConfigProperty(newConfig,
+                ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, serializerClass.getName()));
+
+        Optional.ofNullable(valueDeserializerClass).ifPresent(serializerClass -> overrideConfigProperty(newConfig,
+                ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, serializerClass.getName()));
+
+        setUniqueClientId(newConfig, consumerName, ConsumerConfig.CLIENT_ID_CONFIG);
+        return newConfig;
+    }
+
+    /**
+     * Adapt the properties.
+     * <p>
+     * Subclasses may overwrite this method to set expected configuration values. The default implementation does
+     * nothing.
+     *
+     * @param config The consumer configuration to be adapted.
+     */
+    protected void adaptConfiguration(final Map<String, String> config) {
+
+    }
+
+    /**
+     * Sets the timeout for polling records.
+     * <p>
+     * The default value of this property is {@link #DEFAULT_POLL_TIMEOUT}.
+     *
+     * @param pollTimeoutMillis The maximum number of milliseconds to wait.
+     * @throws IllegalArgumentException if poll timeout is negative.
+     */
+    public final void setPollTimeout(final long pollTimeoutMillis) {
```

Comment:
why is this modeled as a dedicated property? isn't this included in the map that contains the consumer configuration properties?

Where is this property being used?

Reply:
> why is this modeled as a dedicated property? isn't this included in the map that contains the consumer configuration properties?

This is not a Kafka client config but instead an Hono specific property that was once intended to be used by `HonoKafkaConsumer`. This PR does not touch it but only pulls it u into the new base class.

> Where is this property being used?

@calohmn Do you know why it is no longer in use?

Reply:
This was used in the `AbstractAtLeastOnceKafkaConsumer` class.
At one point, that class got replaced by the `HonoKafkaConsumer` class, which doesn't use the `AbstractKafkaConsumerConfigProperties` though. So this is still a TODO, using this property in the `HonoKafkaConsumer` class.

See also the text in #2904:
> The Command Router Kafka command consumer config needs:
> [...]
> specific properties introduced by the Hono consumer classes, like `HonoKafkaConsumer#POLL_TIMEOUT` and the `AsyncHandlingAutoCommitKafkaConsumer#CONFIG[*]` properties
(see also the Hono `KafkaConsumerConfigProperties` class, currently not used for these classes)